src/server/scripts/EasternKingdoms/Karazhan/boss_shade_of_aran.cpp

```diff
@@ -70,560 +71,492 @@ enum SuperSpell
-            {
-                libraryDoor->SetGoState(GO_STATE_ACTIVE);
-                libraryDoor->RemoveGameObjectFlag(GO_FLAG_NOT_SELECTABLE);
-            }
-        }
-
-        void KilledUnit(Unit* /*victim*/) override
+        if (GameObject* libraryDoor = instance->instance->GetGameObject(instance->GetGuidData(DATA_GO_LIBRARY_DOOR)))
         {
-            Talk(SAY_KILL);
+            libraryDoor->SetGoState(GO_STATE_ACTIVE);
+            libraryDoor->RemoveGameObjectFlag(GO_FLAG_NOT_SELECTABLE);
         }
 
-        void JustDied(Unit* /*killer*/) override
-        {
-            Talk(SAY_DEATH);
-
-            instance->SetData(DATA_ARAN, DONE);
+        ScheduleHealthCheckEvent(40, [&]{
+            Creature* ElementalOne = me->SummonCreature(CREATURE_WATER_ELEMENTAL, -11168.1f, -1939.29f, 232.092f, 1.46f, TEMPSUMMON_TIMED_OR_CORPSE_DESPAWN, 90000);
+            Creature* ElementalTwo = me->SummonCreature(CREATURE_WATER_ELEMENTAL, -11138.2f, -1915.38f, 232.092f, 3.00f, TEMPSUMMON_TIMED_OR_CORPSE_DESPAWN, 90000);
+            Creature* ElementalThree = me->SummonCreature(CREATURE_WATER_ELEMENTAL, -11161.7f, -1885.36f, 232.092f, 4.59f, TEMPSUMMON_TIMED_OR_CORPSE_DESPAWN, 90000);
+            Creature* ElementalFour = me->SummonCreature(CREATURE_WATER_ELEMENTAL, -11192.4f, -1909.36f, 232.092f, 6.19f, TEMPSUMMON_TIMED_OR_CORPSE_DESPAWN, 90000);
 
-            if (GameObject* libraryDoor = instance->instance->GetGameObject(instance->GetGuidData(DATA_GO_LIBRARY_DOOR)))
+            if (ElementalOne)
             {
-                libraryDoor->SetGoState(GO_STATE_ACTIVE);
-                libraryDoor->RemoveGameObjectFlag(GO_FLAG_NOT_SELECTABLE);
+                Unit* target = SelectTarget(SelectTargetMethod::Random, 1, 100, true);
+                if (!target)
+                    return;
+
+                DoStartNoMovement(target);
+                ElementalOne->SetInCombatWithZone();
+                ElementalOne->CombatStart(target);
+                ElementalOne->SetFaction(me->GetFaction());
+                ElementalOne->SetUnitMovementFlags(MOVEMENTFLAG_ROOT);
+                ElementalOne->SetModifierValue(UNIT_MOD_RESISTANCE_FROST, BASE_VALUE, 0);
             }
-        }
 
-        void JustEngagedWith(Unit* /*who*/) override
-        {
-            Talk(SAY_AGGRO);
+            if (ElementalTwo)
+            {
+                Unit* target = SelectTarget(SelectTargetMethod::Random, 1, 100, true);
+                if (!target)
+                    return;
+
+                DoStartNoMovement(target);
+                ElementalTwo->SetInCombatWithZone();
+                ElementalTwo->CombatStart(target);
+                ElementalTwo->SetFaction(me->GetFaction());
+                ElementalTwo->SetUnitMovementFlags(MOVEMENTFLAG_ROOT);
+                ElementalTwo->SetModifierValue(UNIT_MOD_RESISTANCE_FROST, BASE_VALUE, 0);
```

Comment:
Removing resistance? should be done in DB
Root flags also in DB

Reply:
Base resistance is sniffed, no?

Reply:
Do you want me to actually add the root flags in DB as well? As they may be incorrect, and didn't seem to work (well) regardless

Reply:
Just checked. DB has Frost immunity for these. So should be sufficient to just remove this